### ISSUE TITLE
TypeError: Cannot set property 'source' of undefined

### ISSUE THREAD

**Issue by Garfonso** [ioBroker/ioBroker.admin#472]
When deleting multiple states from alias.0 I constantly get this error:

```
> 2019-12-02 15:56:54.912  - info: history.0 (23716) disabled logging of alias.0.lamps.Terrasse.state
> 2019-12-02 15:56:54.995  - info: history.0 (23716) disabled logging of alias.0.lamps.Spielzimmer.state
> 2019-12-02 15:56:55.167  - info: history.0 (23716) disabled logging of alias.0.lamps.Spielzimmer.saturation
> 2019-12-02 15:56:55.222  - error: admin.0 (23251) uncaught exception: Cannot set property 'source' of undefined
> 2019-12-02 15:56:55.223  - error: admin.0 (23251) TypeError: Cannot set property 'source' of undefined
>     at objects.getObject (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:6017:72)
>     at (anonymous function).(anonymous function) (/opt/iobroker/node_modules/iobroker.objects-redis/index.js:17:63857)
>     at tryCatcher (/opt/iobroker/node_modules/standard-as-callback/built/utils.js:11:23)
>     at promise.then (/opt/iobroker/node_modules/standard-as-callback/built/index.js:19:49)
>     at process._tickCallback (internal/process/next_tick.js:68:7)
> 2019-12-02 15:56:55.223  - info: admin.0 (23251) terminating http server on port 8081
> 2019-12-02 15:56:55.243  - info: admin.0 (23251) terminating
> 2019-12-02 15:56:55.244  - info: admin.0 (23251) Terminated (NO_ERROR): Without reason
> 2019-12-02 15:56:55.336  - info: history.0 (23716) disabled logging of alias.0.lamps.Spielzimmer.hue
> 2019-12-02 15:56:55.419  - info: history.0 (23716) disabled logging of alias.0.lamps.Spielzimmer.colortemp
> 2019-12-02 15:56:55.546  - info: history.0 (23716) disabled logging of alias.0.lamps.Spielzimmer.color
> 2019-12-02 15:56:55.629  - info: history.0 (23716) disabled logging of alias.0.lamps.Spielzimmer.brightness
> 2019-12-02 15:56:55.732  - info: history.0 (23716) disabled logging of alias.0.lamps.Spiegel_Unten.state
> 2019-12-02 15:56:55.821  - error: host.Server Caught by controller[0]: TypeError: Cannot set property 'source' of undefined
> 2019-12-02 15:56:55.822  - error: host.Server Caught by controller[0]:     at objects.getObject (/opt/iobroker/node_modules/iobroker.js-controller/lib/adapter.js:6017:72)
> 2019-12-02 15:56:55.822  - error: host.Server Caught by controller[0]:     at (anonymous function).(anonymous function) (/opt/iobroker/node_modules/iobroker.objects-redis/index.js:17:63857)
> 2019-12-02 15:56:55.822  - error: host.Server Caught by controller[0]:     at tryCatcher (/opt/iobroker/node_modules/standard-as-callback/built/utils.js:11:23)
> 2019-12-02 15:56:55.822  - error: host.Server Caught by controller[0]:     at promise.then (/opt/iobroker/node_modules/standard-as-callback/built/index.js:19:49)
> 2019-12-02 15:56:55.822  - error: host.Server Caught by controller[0]:     at process._tickCallback (internal/process/next_tick.js:68:7)
> 2019-12-02 15:56:55.822  - info: host.Server instance system.adapter.admin.0 terminated with code 0 (NO_ERROR)
> 2019-12-02 15:56:55.822  - info: host.Server Restart adapter system.adapter.admin.0 because enabled
```

The states have logging (history) enabled and also lovelace, iogo & iqontrol and a smartName (Alexa). Maybe that is connected. 

( pastebin link for better readability: https://pastebin.com/CAEjsck0 )

The number of states it needs for the error to show up varies a lot. Maybe something needs to try and set the state during deletion? Sometimes I can only delete up to 3-5 states, sometimes I can delete 20 or more and all is fine.

**Comment by Apollon77** [comment]
Can you please show the object definition for the affected source object? (Objects-pencil icon, "raw" tab, post JSON from there please

**Comment by Apollon77** [comment]
It seems that the "_id" property of the source object is dofferent then the ID

**Comment by Garfonso** [comment]
Hm. Can't reproduce the error currently. I had a bug in a script that overwrote one alias multiple times instead of creating new ones for different targets. Maybe that was connected then? But even with the bug I can not reproduce the issue right now. Strange. 

Ok, I did upgrade JS-controller from 2.1.0 to 2.1.1 a few minutes ago, but it did happen afterwards, still. But maybe the objects were still created by the old controller version. So seems whatever it was, got fixed in newer version (probably what was wrong with the objects can not happen anymore). Sorry for the inconvenience.

I created a paste with some objects that probably caused the issue in the old version, though: https://pastebin.com/sWdRSSh9 
(and the bug in my script was that all those states would be written to alias.0.Spielzimmer.state)
